settlement-pipelines/src/stake_accounts.rs

```diff
@@ -0,0 +1,69 @@
+use anchor_client::anchor_lang::solana_program::stake_history::StakeHistoryEntry;
+use anyhow::anyhow;
+use solana_sdk::clock::Clock;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::stake::state::StakeStateV2;
+use solana_sdk::stake_history::StakeHistory;
+use validator_bonds_common::stake_accounts::{is_locked, CollectedStakeAccounts};
+
+// prioritize collected stake accounts by:
+// - 1. initialized, non-delegated
+// - 2. deactivating
+// - 3. any non-locked
+// - error if all are locked or no stake accounts
+pub fn prioritize_for_claiming(
+    stake_accounts: &CollectedStakeAccounts,
+    clock: &Clock,
+    stake_history: &StakeHistory,
+) -> anyhow::Result<Pubkey> {
+    let mut non_locked_stake_accounts = stake_accounts
+        .iter()
+        .filter(|(_pubkey, _, stake)| !is_locked(stake, clock))
+        .collect::<Vec<_>>();
+    non_locked_stake_accounts.sort_by(|(_, _, stake_account1), (_, _, stake_account2)| {
```

Comment:
sort_by_cached_key would make way more sense here

Reply:
:+1:  Not being aware of this std lib function. Thanks.